### ISSUE TITLE
Problems with Libtiff_jllv4.5 and PROJ_jll

### ISSUE THREAD

**Issue by disberd** [JuliaPackaging/Yggdrasil#6928]
Hi,

I noticed that a package of mine depending on Proj started breaking during CI on a self-hosted gitlab.

I seem to have pinpointed the issue on the latest version of Libtiff_jll (v4.5) that got released a few hours ago and stops including the libtiff library at version 5.
Now we only find `libtiff.so.6` but apparently PROJ_jll internally still points to `libtiff.so.5`, throwing an error if libtiff.so.5 is not already installed in the system when doing `using PROJ_jll`.

I tried manually addng Libtiff_jll@v4.4 and PROJ goes without error, but it seems I can't specify a compat entry to Libtiff_jll in my Project.toml.

Even though I add Libtiff_jll as a dependency and add "4.4" in the compat I still get version 4.5 when instantiating the project.

Does anybody know how to fix this?

**Comment by eschnett** [comment]
@disberd There was a problem with the initial release of `Libtiff_jll` @4.5.1; it led to a circular dependency. This has been corrected by now. Maybe this was causing your problems? If so, updating to the newest `Libtiff_jll` would help.

The update of `Libtiff_jll` was (apart from the above) a straightforward change. Unless there is an incompatibility in the C sources for `libtiff`, I don't know why its soversion would change.

Regarding compatibility entries: An entry of `4.4` means "`4.4` and later, but less than `5`" (See https://pkgdocs.julialang.org/v1/compatibility/#Tilde-specifiers). Maybe `~4.4` would be correct?

**Comment by disberd** [comment]
Hi @eschnett, Thanks a lot for the prompt reply. Indeed I completely misrememberd the compat synthax. Indeed fixing the version  4.4 with "~4.4" fixes the problem.

Regarding the cyclic dependency, I don't think this solves the problem with PROJ_jll.

The error I get when trying to load PROJ_jll (in a temp env with only PROJ_jll) is the following:
```
julia> using PROJ_jll                                                                                                                
ERROR: InitError: could not load library "/root/.julia/artifacts/8a643038d2adde781829b4467a32afa307e23b51/lib/libproj.so"            libtiff.so.5: cannot open shared object file: No such file or directory                                                              
Stacktrace:                                                                                                                            
[1] dlopen(s::String, flags::UInt32; throw_error::Bool)                                                                                
@ Base.Libc.Libdl ./libdl.jl:117                                                                                                   
[2] dlopen(s::String, flags::UInt32)        
```
I did check inside the artifacts folder of Libtiff_jll 4.5.1 and it only has `libtiff.so.6` as mentioned above, as opposed to version 4.4 that has `libtiff.so.5` and works fine with PROJ_jll

For newer version of PROJ_jll this could be fixed by having a [compat] entry for Libtiff (which is currently unbounded), but that won't fix the problem for existing version of PROJ_jll I guess

**Comment by giordano** [comment]
Fixed by #6930?

**Comment by disberd** [comment]
@giordano yeah it seems to work fine now with the latest PROJ version thanks
